ci/dev/watch.ts

```diff
@@ -1,139 +1,204 @@
-import * as cp from "child_process"
+import { spawn, fork, ChildProcess } from "child_process"
+import del from "del"
+import { promises as fs } from "fs"
 import * as path from "path"
-import { onLine } from "../../src/node/util"
-
-async function main(): Promise<void> {
-  try {
-    const watcher = new Watcher()
-    await watcher.watch()
-  } catch (error: any) {
-    console.error(error.message)
-    process.exit(1)
-  }
+import { CompilationStats, onLine, OnLineCallback, VSCodeCompileStatus } from "../../src/node/util"
+
+interface DevelopmentCompilers {
+  [key: string]: ChildProcess | undefined
+  vscode: ChildProcess
+  vscodeWebExtensions: ChildProcess
+  codeServer: ChildProcess
+  plugins: ChildProcess | undefined
 }
 
 class Watcher {
-  private readonly rootPath = path.resolve(__dirname, "../..")
-  private readonly vscodeSourcePath = path.join(this.rootPath, "vendor/modules/code-oss-dev")
+  private rootPath = path.resolve(process.cwd())
```

Comment:
nit: since we moved `vscodeSourcePath` any reason not to move `rootPath` as well? That way `paths` has all paths.

Reply:
Unfortunately JS doesn’t let you reference earlier object properties while still defining entries.

Reply:
Ahhhh got it! Disregard 👍